internal/controller/predicates/predicates.go

```diff
@@ -12,10 +12,27 @@ type GrafanaPodRecreatedPredicate struct {
 	predicate.Funcs
 }
 
-func (GrafanaPodRecreatedPredicate) Delete(e event.DeleteEvent) bool {
-	if e.Object != nil && strings.Contains(e.Object.GetName(), "grafana") && e.Object.GetNamespace() == "monitoring" {
+// When a grafana pod is recreated, we want to trigger a reconciliation.
+func (GrafanaPodRecreatedPredicate) Create(e event.CreateEvent) bool {
+	if e.Object != nil &&
+		strings.Contains(e.Object.GetName(), "grafana") &&
+		e.Object.GetNamespace() == "monitoring" &&
+		// Ensure we don't trigger on the grafana permissions pods or grafana multi-tenant proxy
+		e.Object.GetLabels()["app.kubernetes.io/instance"] == "grafana" {
 		return true
 	}
 
 	return false
 }
+
+func (GrafanaPodRecreatedPredicate) Delete(e event.DeleteEvent) bool {
+	return false
```

Comment:
Fixes the évent handler predicate to only act on grafana restarts and.not any update on pods

Reply:
Also add this as a comment

Reply:
Can you make a suggestion?

Reply:
```suggestion
	// Prevent to react on grafana restarts.
	return false
```